src/reducers/facetconfig.ts

```diff
@@ -60,7 +63,35 @@ export const lastId = (state: FacetConfigReducer): number =>
 export const getById = (id: number, state: FacetConfigReducer): NormalizedFacetConfig | undefined =>
     state.find(config => config.id === id);
 
+export const denormalizeFacetConfig = (config: NormalizedFacetConfig): FacetConfig => {
+    config = { ...config };
+
+    for (const [idx, referencePath] of config.referencePaths.entries()) {
+        config.paths[idx] = mendPath(referencePath);
+    }
+
+    delete config.referencePaths;
+    delete config.id;
+    return config;
+};
+
+export const composeFacets = (facetConfigs: NormalizedFacetConfig[]): FacetConfig[] =>
```

Comment:
also rename to `denormalize` (or rename `denormalizeFacetConfig` to `composeFacetConfig`)